### ISSUE TITLE
In order to make code easier to understand I want remove semi-arbitrary delay while hiding the loading indicator

### ISSUE THREAD

**Issue by vladimir-8** [coopcycle/coopcycle-app#1670]
**I want** `something to be added or changed`
A clear and concise description of what is to be added or changed in the software

Based on the tests I did, it seems that after removing the delay for a spinner, everything still works as expected. I hope that this delay is not needed anymore. However, I’m not entirely confident it will not put back this issue: https://github.com/coopcycle/coopcycle-app/issues/1265. 

I propose to make this change in a few steps:

- [ ] 1. Ask a few couriers to manually set it via the feature flags

- [ ] 2. Set it as a default option, while still allowing to revert it via a feature flag
https://github.com/coopcycle/coopcycle-app/blob/a67eee6cc3cd6a6b51968dc6340744e177a2f213/src/redux/App/reducers.js#L100
https://github.com/coopcycle/coopcycle-app/blob/a67eee6cc3cd6a6b51968dc6340744e177a2f213/src/redux/App/selectors.js#L145

- [ ] 3. Remove the feature flag and clean up the code

**Whereas currently** `this happens`
A clear and concise description of what happens now

I suspect that the issue with displaying Modals on iOS (for example, maintenance message: https://github.com/coopcycle/coopcycle-app/issues/1669) is happening due to this extra delay that we added before hiding a spinner: https://github.com/coopcycle/coopcycle-app/commit/866fd9a8ce4c65c52ccb5d8270d380df2882d87a

https://coopcycle.slack.com/archives/G01JR0YF5V3/p1704250963518979

**Comment by vladimir-8** [comment]
Currently, the delay is applied only in the Courier interface on iOS: https://github.com/coopcycle/coopcycle-app/blob/master/src/redux/App/selectors.js#L176-L182 Removing it causes the app to be stuck (invisible Modal overlay) on task completion. We need to figure out why that happens before we can remove it. That has low priority, however.
